cpp/oneapi/dal/algo/pca/backend/gpu/train_kernel_svd_impl_dpc.cpp

```diff
@@ -0,0 +1,207 @@
+    ONEDAL_ASSERT(data.get_dimension(0) == data.get_dimension(1),
+                  "dataelation matrix must be square");
+    ONEDAL_ASSERT(data.get_dimension(0) > 0);
+    const std::int64_t column_count = data.get_dimension(0);
+
+    auto eigvecs = pr::ndarray<Float, 2>::empty({ component_count, column_count });
+    auto eigvals = pr::ndarray<Float, 1>::empty(component_count);
+
+    auto host_data = data.to_host(q, deps);
+    pr::sym_eigvals_descending(host_data, component_count, eigvecs, eigvals);
+
+    return std::make_tuple(eigvecs, eigvals);
+}
+
+template <typename Float>
+auto centered_data(sycl::queue& q,
+                   const pr::ndview<Float, 2>& data,
+                   const bk::event_vector& deps = {}) {
+    ONEDAL_PROFILER_TASK(centered_data, q);
+    ONEDAL_ASSERT(data.has_data());
+    ONEDAL_ASSERT(data.get_dimension(1) > 0);
+    const std::int64_t column_count = data.get_dimension(1);
+    const std::int64_t row_count = data.get_row_count();
+
+    pr::ndarray<Float, 2> data_copy =
+        pr::ndarray<Float, 2>::copy(data, { row_count, column_count });
+
+    auto event = q.submit([&](sycl::handler& cgh) {
+        auto data_acc = data_copy.get_mutable_data();
+
+        cgh.parallel_for(sycl::range<1>(column_count), [=](sycl::id<1> idx) {
+            Float sum = 0.0;
+            for (std::int64_t i = 0; i < row_count; i++) {
+                sum += data_acc[i * column_count + idx[0]];
+            }
+            Float mean = sum / row_count;
+            for (std::int64_t i = 0; i < row_count; i++) {
+                data_acc[i * column_count + idx[0]] -= mean;
+            }
+        });
+    });
+
+    return std::make_tuple(data_acc, event);
+}
+
+template <typename Float>
+auto svd_decomposition(sycl::queue& q,
+                       pr::ndview<Float, 2>& data,
+                       std::size_t row_count,
+                       std::size_t column_count) {
+    auto U = pr::ndarray<Float, 2>::empty({ row_count, row_count });
+    auto S = pr::ndarray<Float, 1>::empty({ column_count });
+    auto V_T = pr::ndarray<Float, 2>::empty({ column_count, column_count });
+
+    //auto result = mkl::lapack::gesvd(queue, uplo, trans, n, k, alpha, a, lda, beta, c, ldc, deps);
+    Float* data_ptr = data.get_mutable_data();
+    Float* U_ptr = U.get_mutable_data();
+    Float* S_ptr = S.get_mutable_data();
+    Float* V_T_ptr = V_T.get_mutable_data();
+
+    const std::size_t min_dim = std::min(row_count, column_count);
+    const std::size_t max_dim = std::max(row_count, column_count);
+    const std::size_t num_iters = 100;
+
+    q.submit([&](sycl::handler& cgh) {
+        auto data_acc = sycl::accessor(data_ptr, cgh, sycl::read_write);
+        auto U_acc = sycl::accessor(U_ptr, cgh, sycl::write);
+        auto S_acc = sycl::accessor(S_ptr, cgh, sycl::write);
+        auto V_T_acc = sycl::accessor(V_T_ptr, cgh, sycl::write);
+
+        cgh.parallel_for(sycl::range<1>(1), [=](sycl::id<1> idx) {
```

Comment:
It will result in just horrible performance. 
1) You can use `cgh.single_task` [link](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_queue_interface)
2) What about using more items?